allennlp/common/from_params.py

```diff
@@ -564,6 +564,17 @@ def from_params(
         else:
             # This is not a base class, so convert our params and extras into a dict of kwargs.
 
+            # If there was a "type" given, we probably didn't want to end up here.
+            if "type" in params:
```

Comment:
This check isn't sufficient.  If the test had done `A.from_params({"some": "dict"})`, you wouldn't get here.  I'll look into getting a better fix for this.